src/migrateShards.ts

```diff
@@ -0,0 +1,30 @@
+import {ClusterSizeCheckResponse, DefaultResponse} from './utils/handlerResponses';
+import {getShardInfo, migrateShards} from './elasticsearch/elasticsearch';
+import {ElasticsearchNode, Move, ShardCopy} from './elasticsearch/types'
+
+export async function handler(event: ClusterSizeCheckResponse): Promise<DefaultResponse> {
+    const oldestElasticsearchNode: ElasticsearchNode = event.oldestElasticsearchNode;
+    const newestElasticsearchNode: ElasticsearchNode = event.newestElasticsearchNode;
+    return getShardInfo(oldestElasticsearchNode.ec2Instance.id)
+        .then(response => moveInstructions(response, oldestElasticsearchNode, newestElasticsearchNode))
+        .then(moves => migrateShards(moves, oldestElasticsearchNode));
+}
+
+export function moveInstructions(response, oldNode: ElasticsearchNode, newNode: ElasticsearchNode): Move[] {
```

Comment:
minor: if this is only being exported for tests then exporting as part of an underscore object is a good way to signal this, and hopefully try and keep it 'private'"

`export const _ = { moveInstructions };`